plugin-build/build.gradle.kts

```diff
@@ -87,15 +87,23 @@ val unzipKotlinCompiler by tasks.registering(Copy::class) {
 
 val testPluginKotlinc by tasks.registering(RunTestExecutable::class) {
     dependsOn(unzipKotlinCompiler, tasks.shadowJar)
-    executable(file("${unzipKotlinCompiler.get().destinationDir}/kotlinc/bin/kotlinc"))
+
     args(
         listOf(
             "$rootDir/src/test/resources/hello.kt",
             "-Xplugin=${tasks.shadowJar.get().archiveFile.get().asFile.absolutePath}",
             "-P",
-            "plugin:detekt-compiler-plugin:debug=true"
         )
     )
+
+    if (org.apache.tools.ant.taskdefs.condition.Os.isFamily("windows")) {
+        executable(file("${unzipKotlinCompiler.get().destinationDir}/kotlinc/bin/kotlinc.bat"))
+        args("\"plugin:detekt-compiler-plugin:debug=true\"")
+    } else {
+        executable(file("${unzipKotlinCompiler.get().destinationDir}/kotlinc/bin/kotlinc"))
+        args("plugin:detekt-compiler-plugin:debug=true")
```

Comment:
nit: I would set these strings on a common variable and add the `.bat` and `\"` for the windows version. This way we can keep them aligned a bit easier.